x-pack/plugin/search-business-rules/src/main/java/org/apache/lucene/search/CappedScorer.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.apache.lucene.search;
+
+import java.io.IOException;
+
+public class CappedScorer extends Scorer
+{
+    Scorer delegate;
+    private float maxScore;
+    
+
+  public CappedScorer(Weight weight, Scorer delegate, float maxScore) {
+      super(weight);
+      this.delegate = delegate;
+      this.maxScore = maxScore;
+  }
+
+  @Override
+  public DocIdSetIterator iterator() {
+      return delegate.iterator();
+  }
+
+  @Override
+  public float getMaxScore(int upTo) throws IOException {
+      return maxScore;
```

Comment:
Let's return `Math.min(maxScore, delegate.getMaxScore(upTo))`, it might sometimes return a lower bound? Note that we would also need to delegate `advanceShallow`.